core/src/smartcontracts/wasm.rs

```diff
@@ -78,6 +79,66 @@ impl<'a, W: WorldTrait> State<'a, W> {
 
         Ok(())
     }
+
+    fn validate_instruction(
```

Comment:
Feels like this should be 
```
impl ValidatorTrait for Validator{ 
    type Target = Instruction;
    
}
```

but I suppose that this can wait.

Reply:
I think this can wait.